controllers/clusteroperator.go

```diff
@@ -179,7 +179,7 @@ func (r *ProvisioningReconciler) updateCOStatus(newReason StatusReason, msg, pro
 		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionFalse, string(newReason), progressMsg))
 	case ReasonInvalidConfiguration, ReasonDeployTimedOut:
 		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionTrue, string(newReason), msg))
-		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonEmpty), ""))
+		v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonAsExpected), "FIXME: are we really available here?"))
```

Comment:
I think Degraded makes sense

Reply:
I'm fine with Degraded=True here.  I'm not clear on Available=True.  What functionality is the operator available to fulfil when it has ReasonInvalidConfiguration or ReasonDeployTimedOut?

Reply:
We looked into several options detailed here ; https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md#not-in-use-slo-behaviors

There seems to be no defined behavior when a CO needs to convey that the Operator is running, is non-functional and that is expected.

Reply:
Neither `ReasonInvalidConfiguration` nor `ReasonDeployTimedOut` are "expected non-functional", are they?  That is `ReasonUnsupported`, which I also touch in this PR, but it is not what this thread/line is about.

Reply:
Agreed. Giving context for you question if Available=True is required.

Reply:
`Available` is definitely require.  In master, `ReasonInvalidConfiguration` and `ReasonDeployTimedOut` are `Available=True` `Degraded=False`.  My naive reaction to the reason strings has me suspecting the might actually call for `Available=False` `Degraded=False`.  Can you list and services available (but degraded) under these conditions?

Reply:
> I'm fine with Degraded=True here. I'm not clear on Available=True. What functionality is the operator available to fulfil when it has ReasonInvalidConfiguration or ReasonDeployTimedOut?

The Operator is Available because it is still watching its resources. If the configuration changes to a valid value for example, then the operator is able to get itself out of the Degraded state. That is how I view the Available=True in this scenario.